### PR TITLE
(MINOR) Improved Remote Manifest handling

### PR DESCRIPTION
**Pull request by gpeacock**
Modifies Ingredient::from_ methods  to return a MANIFEST_INACCESSIBLE verification_status code 
the validation status url will include the required manifest url.
when a cloud manifest needs to be retrieved but cannot be. The ingredient is always created and no other error is returned.
If there is an embedded manifest, no attempt is made to read the cloud manifest and MANIFEST_INACCESSIBLE status injected.
fetch remote manifests is not currently implemented for stream and memory versions of Ingredient::from_*
If fetch_remote_manifests is disabled, manifest.inaccessible validation status will be added whenever a cloud url is needed.
if fetch_remote_manifests is enabled, and the request fails, a MANIFEST_INACCESSIBLE verification_status code will be added.

The returned URL can be used to fetch the manifest independently.

Two new Ingredient methods have been added to allow validating generating and validating an ingredient if you have the asset and the manifest.
if a manifest (.c2pa) is available either from a sidecar or a retrieved externally from the cloud, these can be used to create a fully validated Ingredient.

Adds Ingredient::from_manifest_and_asset_bytes_async
Adds Ingredient::from_manifest_and_asset_stream_async

### CODE REVIEW

**Comment by dkozma** [comment]
@gpeacock I tried integrating this, however I found when adding an ingredient, the ingredient's manifest would have multiple validation status errors. I tried to create a test in c2pa-rs to reproduce this to make sure it wasn't anything weird going on with the c2pa-js code. You can see the test code here: https://github.com/contentauth/c2pa-rs/tree/dkozma/remote_ingredient_wasm_test (run with `make test-wasm-web`).

File: sdk/src/ingredient.rs [hunk 1]
```diff
@@ -1148,10 +1244,57 @@ mod tests {
...
+
+    #[cfg_attr(not(target_arch = "wasm32"), actix::test)]
+    #[cfg_attr(target_arch = "wasm32", wasm_bindgen_test)]
+    async fn test_jpg_cloud_from_memory() {
+        let image_bytes = include_bytes!("../tests/fixtures/cloud.jpg");
+        let format = "image/jpeg";
+        let ingredient = Ingredient::from_memory_async(format, image_bytes)
+            .await
+            .expect("from_memory_async");
+        //println!("ingredient = {ingredient}");
```
Comment (dkozma) [inline sdk/src/ingredient.rs:1264]: ✂️

File: sdk/src/ingredient.rs [hunk 2]
```diff
@@ -1148,10 +1244,57 @@ mod tests {
...
+            format,
+            asset_bytes,
+        )
+        .await
+        .unwrap();
+        #[cfg(target_arch = "wasm32")]
+        web_sys::console::debug_2(
+            &"ingredient_from_memory_async:".into(),
+            &ingredient.to_string().into(),
+        );
```
Comment (dkozma) [inline sdk/src/ingredient.rs:1294]: ✂️

**Review by dkozma — APPROVED** [review]
Couple debug messages you may want to remove from tests but overall LGTM.

**Review by dyro — APPROVED** [review]
⚡

File: sdk/src/ingredient.rs [hunk 3]
```diff
@@ -967,6 +981,88 @@ impl Ingredient {
...
+        let manifest_bytes: Vec<u8> = manifest_bytes.into();
+        // generate a store from the buffer and then validate from the asset path
+        let result = match Store::from_jumbf(&manifest_bytes, &mut validation_log) {
+            Ok(store) => {
+                // verify the store
+                //todo, change this when we have a stream version of verify
+                let mut buf: Vec<u8> = Vec::new();
+                stream.rewind()?;
+                stream.read_to_end(&mut buf).map_err(Error::IoError)?;
+                Store::verify_store_async(&store, &buf, &mut validation_log)
```
Comment (mauricefisher64) [inline sdk/src/ingredient.rs:1035]: This wont be needed once my bmff_v2 branch is merged.

**Pull request merged**